Description: libconfig 1.5 ABI Change
 Libconfig changes ABI with version 1.5, setting operator[std::string] is removed
Author: Tobias Frost <tobi@debian.org>
Last-Update: 2015-08-05
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/configuration/CConfigHelper.h
+++ b/src/configuration/CConfigHelper.h
@@ -312,7 +312,7 @@
             libconfig::Setting & set =
                 Registry::Instance().GetSettingsForObject(cfgpath);
 
-            store = set[setting][index];
+            store = set[setting.c_str()][index];
             return true;
 #if 0
         } catch (libconfig::SettingNotFoundException &e) {
@@ -349,7 +349,7 @@
             libconfig::Setting & set =
                 Registry::Instance().GetSettingsForObject(cfgpath);
 
-            store = (const char *)set[setting][index];
+            store = (const char *)set[setting.c_str()][index];
             return true;
 #if 0
         } catch (libconfig::SettingNotFoundException &e) {
